src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -492,16 +538,18 @@ function copy_to_coupled_boundary!(boundary_condition::BoundaryConditionCoupled{
     for cell in cells
         i_node = i_node_start
         j_node = j_node_start
-
-        for i in eachnode(solver)
-            for v in 1:size(u, 1)
-                boundary_condition.u_boundary[v, i, cell] = u[v, i_node, j_node,
-                                                              linear_indices[i_cell,
-                                                                             j_cell]]
-            end
+        element_id = linear_indices[i_cell, j_cell]
+
+        for element_id in eachnode(solver)
+            x = @view cache.elements.node_coordinates[:, i_node, j_node,
+                                                      linear_indices[i_cell, j_cell]]
+            u_node = u[:, i_node, j_node, linear_indices[i_cell, j_cell]]
```

Comment:
```suggestion
            x = get_node_coords(node_coordinates, equations, solver, i_node, j_node,
                                linear_indices[i_cell, j_cell])
            u_node = get_node_vars(u, equations, solver, i_node, j_node,
                                   linear_indices[i_cell, j_cell]]
```
This will extract an `SVector` (without allocations). This has several advantages. For example, users can't accidentally modify `x` or `u` in the converter.

But more importantly, you can do all kinds of operations with `SVectors` without causing allocations. With views, you can't really do anything. A view is based on a pointer to the same memory where the big array is stored. That means you can't do any operations with a view without writing the content somewhere (i.e. allocating a vector). For example, this doesn't allocate, it's just a view:
```julia
julia> x = rand(2, 2)
2×2 Matrix{Float64}:
 0.370773  0.904133
 0.547909  0.496213

julia> v = view(x, :, 1)
2-element view(::Matrix{Float64}, :, 1) with eltype Float64:
 0.3707729787028192
 0.547909002762257
```
This, however, yields a vector, and therefore has to allocate memory:
```julia
julia> 2v
2-element Vector{Float64}:
 0.7415459574056384
 1.095818005524514
```